shared/js/sk_key_utils.js

```diff
@@ -0,0 +1,367 @@
+    var observer = null;
+
+    return {
+      init: init
+    };
+
+    /**
+     * Used to initialize the library
+     * @example
+     * softKeyModule.init();
+     *
+     * @param N/A
+     * @return void 0
+     */
+    function init() {
+
+      // Calling destroy to destroy previously initialized variables
+      destroy();
+
+      // Selecting the softkey DOM element
+      var softKey = document.querySelector('[role="softkey"]');
+      if (!softKey) {
+        // If null, return
+        return;
+      }
+
+      // Container will be created to handle options
+      // menu and soft keys
+      createSoftKeyContainer();
+
+      for (var k in sk_keys) {
+        if (sk_keys.hasOwnProperty(k)) {
+
+          // Selecting the button using accesskey
+          sk_keys[k] =
+                  softKey.querySelector('[accesskey="' + SK_KEYS_MAPPING[k] + '"]');
+
+          if (sk_keys[k]) {
+
+            // Softkeys will be created for NON-Touch UI only
+            // A #IFDEF check is required
+            createSoftKeyButtons(sk_keys[k], k);
+
+            // Adding Event Listener
+            sk_keys[k].addEventListener('click',
+                    handleSoftKey,
+                    false);
+          }
+        }
+      }
+
+      // Start observing DOM change
+      observeDOMChange(softKey);
+    }
+
+    /**
+     * Used to create soft key container
+     * @example
+     * createSoftKeyContainer();
+     *
+     * @param N/A
+     * @return void 0
+     */
+    function createSoftKeyContainer() {
+      var sk_button_container = null;
+
+      // Parent container
+      sk_container = document.createElement('div');
+      sk_container.dataset.subtype = 'menu-parent';
+      sk_container.tabIndex = -1;
+      sk_container.setAttribute('role', 'dialog');
+
+      sk_button_container = document.createElement('div');
+      sk_button_container.classList.add('sk-button-container');
+      sk_container.appendChild(sk_button_container);
+
+      document.body.appendChild(sk_container);
+    }
+
+    /**
+     * Used to create soft key buttons
+     * @example
+     * createSoftKeyButtons(element, 'LSK');
+     *
+     * @param {object, string}
+     *        key: Key which will be pressed
+     *        keyType: Key type ie. LSK, RSK, CSK. 
+     *                 Required for adding styles
+     * @return void 0
+     */
+    function createSoftKeyButtons(key, keyType) {
+      var button = document.createElement('div');
+      button.textContent = key.textContent;
+      button.classList.add('sk-buttons', keyType);
+      sk_container
+              .querySelector('.sk-button-container')
+              .appendChild(button);
+
+      // Hide the buttons for Non-Touch UI
+      key.classList.add('sk-hide-button');
+    }
+
+    /**
+     * Used to handle button press
+     * @example
+     * handleSoftKey();
+     *
+     * @param N/A
+     * @return void 0
+     */
+    function handleSoftKey() {
```

Comment:
I don't want to think about contextmenus yet in this review, so I'm ignoring most of this function.

You need event handlers to handle keydown events for the soft keys, and when those occur you probably can just call .click() on the elements that the softkeys map to.